components/DonatedItemsList.tsx

```diff
@@ -34,14 +38,22 @@ const DonatedItemsList: React.FC<DisplayDonatedItemsProps> = ({ userId }) => {
   }, []);
 
   const handleDeleteSuccess = (deleteItemId: number) => {
+    const deletedItem = storeItems.find((item) => item.id === deleteItemId);
+    setDeletedItemName(deletedItem?.item_name || 'Unknown name');
     setStoreItems(storeItems.filter((item) => item.id !== deleteItemId));
+    setDisplayDeletedItemMessage(true);
```

Comment:
We only need the one state here.

I know I'm always going on about 'not making our types week' but this is one of those situations where you bend the patterns:

If the value is potentially `undefined` then we already have our boolean evaluation there.
Just give the `deleteItemName` a value when it needs one and only display when it's there.

```ts
const [deletedItemName, setDeletedItemName] =
    useState<string | undefined>(undefined);
    
     // etc....
    
    const handleDeleteSuccess = (deleteItemId: number) => {
    const deletedItem = storeItems.find((item) => item.id === deleteItemId);
     
    if (deletedItem){
      setDeletedItemName(deletedItem.item_name)
    }
    setStoreItems(storeItems.filter((item) => item.id !== deleteItemId));
  };
```